website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+Binding is a term you will hear a lot with OpenGL, and in these tutorials too. Binding essentially means updating the current OpenGL state with the given object. So, in this example, when we bind the VAO, we update OpenGL's state so that the VAO we are currently using is now the VAO we have just created. If you don't bind an object before updating/using it, you will use or update the previously bound object instead, which can cause problems and headaches later.
+
+We're now ready to give the VAO some data. But first, we need a **vertex buffer**.
+
+## Vertex Buffer Objects (VBOs)
+Here is the part where we actually give the GPU some data it can work with. For 95% of graphics applications, a vertex buffer is **required** to display anything on screen.
+
+### What is a vertex buffer?
+Before we can continue, it's handy to know what a vertex buffer is, and what it does.
+
+Let's define both vertex, and buffer.
+
+* Vertex - A point where two lines, or edges meet.
+* Buffer - A region in memory that can be accessed to be written to, or read from.
+
+Knowing these definitions should give us a good idea of what a vertex buffer is. It's a region of memory, on the GPU, that stores points, which will get **rasterized** on-screen (usually in the form of triangles).
+
+### Vertex data
+Let's create the data that we will fill the vertex buffer with!
+
+In your `OnLoad` function, just after you call `_gl.BindVertexArray()`, add the following:
+
+```cs
+float[] vertices =
+{
+     0.5f,  0.5f, 0.0f,
+     0.5f, -0.5f, 0.0f,
+    -0.5f, -0.5f, 0.0f,
+    -0.5f,  0.5f, 0.0f
+};
+```
+
+### What makes up a quad?
+In modern graphics programming, you are expected to use triangles. While you may see options for quads, or lines, these are mostly deprecated.
+
+Therefore, a quad is made of two right-angle triangles. This can best be seen if we view the result in **wireframe** mode.
+
+![Wireframe quad](../../../images/opengl/chapter1/wireframe-quad.png)
+
+In the image, you can also see where the four vertices go in relation to the quad. While you won't *usually* be defining vertices yourself, it's still handy to know how it works.
+
+### Creating the buffer
+Now that we've got our vertex data, let's create the buffer!
+
+At the top of your class, add the following:
+
+```cs
+private static uint _vbo;
+```
+
+Then, in your `OnLoad` method, under where the vertices are defined, add:
+
+```
+_vbo = _gl.GenBuffer();
+_gl.BindBuffer(BufferTargetARB.ArrayBuffer, _vbo);
+```
+
+Much like the VAO, we need to first generate the buffer, and then bind it. Unlike vertex arrays, however, buffers need to be bound to a **target**. This allows you to bind buffers to different targets at the same time. Some of the more common targets include:
+
+* `ArrayBuffer` - The vertex buffer target (which is what we're using here).
+* `ElementArrayBuffer` - The element array buffer target (which you'll see later).
```

Comment:
If you wish to make this more specific, here's [the doc on binding targets](https://www.khronos.org/opengl/wiki/Buffer_Object#General_use).

In summary:
- `ArrayBuffer`: Note that `glVertexAttribPointer()` doesn't have any way to specify which VBO to target! That's because it targets what's currently bound to `ArrayBuffer`.
- `ElementArrayBuffer`: Uhhhh this one's a bit of a pain. Unlike the other buffer targets, this one is not stored as a "general OpenGL state", but rather is part of the currently bound VAO. `ElementArrayBuffer` controls "which buffer to use as an index buffer for the currently bound VAO".

Reply:
I personally think this is a bit too specific to put in this tutorial, however I could add a note to the shader attributes section with the `glVertexAttribPointer` information, as it seems useful to know.